### PR TITLE
Remove solution options OOP serialization

### PR DESCRIPTION
**Pull request by tmat**
It's no longer necessary to serialize options as part of solution snapshot as they are now passed along RPC calls.

### CODE REVIEW

**Review by JoeRobich — APPROVED** [review]

File: src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs [hunk 1]
```diff
@@ -568,10 +562,9 @@ private async Task ValidateChecksumAsync(Checksum checksumFromRequest, Solution
                     return;
                 }
 
-                var (solutionInfo, options) = await _assetProvider.CreateSolutionInfoAndOptionsAsync(checksumFromRequest, _cancellationToken).ConfigureAwait(false);
+                var solutionInfo = await _assetProvider.CreateSolutionInfoAsync(checksumFromRequest, _cancellationToken).ConfigureAwait(false);
                 var workspace = new AdhocWorkspace(_hostServices);
                 workspace.AddSolution(solutionInfo);
-                workspace.SetCurrentSolution(s => s.WithOptions(options), WorkspaceChangeKind.SolutionChanged);
 
                 await TestUtils.AssertChecksumsAsync(_assetProvider, checksumFromRequest, workspace.CurrentSolution, incrementalSolutionBuilt).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs:569]: for some reason, i thought this was more involved in this file :D

File: src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs [hunk 2]
```diff
@@ -72,12 +72,6 @@ public async Task<Solution> CreateSolutionAsync(Checksum newSolutionChecksum)
                         Contract.ThrowIfFalse(solution.Id == newSolutionInfo.Id && solution.FilePath == newSolutionInfo.FilePath);
                     }
 
-                    if (oldSolutionChecksums.Options != newSolutionChecksums.Options)
-                    {
-                        var newOptions = await _assetProvider.GetAssetAsync<SerializableOptionSet>(newSolutionChecksums.Options, _cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs:77]: do we even need SerializableOptionSet anymore?

File: src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs (hunk 1 above)
Comment (tmat) [reply src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs:569]: The complexity is hidden behind the SerializableOptionSet. Still needs to be cleaned up, but it's more complex ;)

File: src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs (hunk 2 above)
Comment (tmat) [reply src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs:77]: Yes, still need it. The beast has a lot of tentacles. Severing one at a time.

**Pull request merged**